core/src/main/java/org/testcontainers/dockerclient/DockerClientProviderStrategy.java

```diff
@@ -104,6 +121,71 @@ public DockerClient getClient() {
         return dockerClient;
     }
 
+    /**
+     * TODO we should consider moving this to docker-java at some point
+     */
+    @UnstableAPI
+    protected boolean test() {
+        TransportConfig transportConfig = getTransportConfig();
+        URI dockerHost = transportConfig.getDockerHost();
+
+        Callable<Socket> socketProvider;
+        SocketAddress socketAddress;
+        switch (dockerHost.getScheme()) {
```

Comment:
Given this switch statement, I would personally find it more intuitive, to implement `test()` in the individual strategy classes directly. However, given that this is an internal change for now and we consider moving it to `docker-java`, we can of course go with this approach for now.

Reply:
It is `protected` so any impl can override it if switch doesn't cover their scheme (note that it passes the test on an unknown value), but _most_ of (basically all) strategies will benefit from this impl and can reuse it, so why would we duplicate it in every strategy? :)

Reply:
Duplication would be bad, I agree. Whenever I see a switch statement, I just wonder what the corresponding OOP code would look like. Since we have some classes that represent some of the switch cases (such as `NpipeSocketClientProviderStrategy` or `UnixSocketClientProviderStrategy`) I think it would not be unintuitive to have them encapsulate the code specific to their protocols. However, from the switch-case we also see that some strategies would share code and adding different and more abstractions might very well make the implementation more compicate.